tests/api.c

```diff
@@ -66255,10 +66255,28 @@ static const char* apitest_res_string(int res)
 #ifndef WOLFSSL_UNIT_TEST_NO_TIMING
 static double gettime_secs(void)
 {
-    struct timeval tv;
-    LIBCALL_CHECK_RET(gettimeofday(&tv, 0));
+    #if defined(_MSC_VER) && defined(_WIN32)
+    {
+        /* there's no gettimeofday for Windows, so we'll use system time */
```

Comment:
Ah, good idea. But after going down that road I realized `XTIME()` is only granular to the level of whole seconds. In contrast, the `double gettime_secs()` is a `double` as it has 5 decimal places of fractional seconds.

I've updated this PR to instead pull out the `TimeNowInMilliseconds()` from tls13.c and make it a `WOLFSSL_API` in the `internal.h` file, thus allowing [gettime_secs() in api.c to use TimeNowInMilliseconds()](https://github.com/wolfSSL/wolfssl/commit/d8fe8b914080fe2488837e02211b22156dc7cf90). 

We still lose 2 digits of precision, but I wonder how much more granular the measurements need to be than milliseconds? This change also results in `gettime_secs()` being available for many different platforms,

Reply:
For greater precision it looks like there is the macro XTIME_MS(). The suggestion was more to avoid duplication of code if we had a way to collect the number of seconds/microseconds already. If a large refactor or new public API is needed to achieve that then please just use the initial approach of adding a new ifelse macro case in tests/api.c.

Reply:
Can you help me understand how to use `XTIME_MS()`? It appears it is defined as `m2mb_xtime_ms((tl))` and [only in wc_port.h](https://github.com/wolfSSL/wolfssl/blob/70c362f680d06336267c0139f8fb2fd40790f9e4/wolfssl/wolfcrypt/wc_port.h#L995) when gated with `WOLFSSL_TELIT_M2MB` and `WOLFSSL_TLS13`.  The [implementation of m2mb_xtime_ms](https://github.com/wolfSSL/wolfssl/blob/70c362f680d06336267c0139f8fb2fd40790f9e4/wolfcrypt/src/wc_port.c#L3211) appears to be very platform specific, using the `m2mb_rtc_open("/dev/rtc0", 0)`.

I like the multi-platform implementation of `TimeNowInMilliseconds()` as having a common global millisecond timer for all platforms would be very handy, and not specific to only TLS1.3.

It's not so much that I _wanted_ a new public API, but that was not only way I could see how to make it visible in `api.c`. I probably don't understand the wolfSSL visibility macros as well as I'd like.

Given this, how would you prefer that I proceed? Back to the initial approach adding Windows support? Do you think it is possible to get this working without being a `WOLFSSL_API` function?

Reply:
No problem, it was a valid idea. It may be valuable to have a wolfSSL time library at some point.

For now in this PR I've reverted to the initial approach as requested.